pkg/kv/iterators.go

```
@@ -253,24 +253,29 @@ type PartitionIterator struct {
 }
 
 func NewPartitionIterator(ctx context.Context, store Store, msgType protoreflect.MessageType, partitionKey string) (*PartitionIterator, error) {
```

In that case - so why do we need to return an error?